packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -482,8 +518,30 @@ export async function visualizeChildData(
  * Determines whether or not the provided value is an {@link ISharedObject}, for the purposes of this library.
  * @remarks Implemented by checking for the particular properties / methods we use in this module.
  */
-function isSharedObject(value: Partial<ISharedObject>): value is ISharedObject {
+function isSharedObject(value: unknown): value is ISharedObject {
 	return (
-		value.id !== undefined && value.attributes?.type !== undefined && value.on !== undefined
+		(value as SharedObject).id !== undefined &&
+		(value as SharedObject).attributes?.type !== undefined &&
+		(value as SharedObject).on !== undefined
 	);
 }
+
+/**
+ * Determines whether or not the provided value is an {@link DataObject} using `instanceof`, for the purposes of this library.
+ * @remarks `instanceof` is preferred over checking specific properties or methods, because a version mix-up with
+ * @remarks {@link @fluidframework/aqueduct#} is unlikely in devtools and end-user applications.
+ * @remarks `value` is type casted as a workaround to access the `root` property of the {@link DataObject} (private).
```

Comment:
Rewording because I don't think it's really _preferred_, it's just the decision we went with, and its reasoning.

The last line could make sense next to the line it affects.

```suggestion
 * @remarks
 * Uses `instanceof` over checking specific properties or methods, because we decided that a version mix-up with
 * @fluidframework/aqueduct is unlikely between devtools and end-user applications, and we don't support it anyway.
```